atc/engine/check_delegate.go

```diff
@@ -111,25 +118,32 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 		}
 	}
 
-	end, err := scope.LastCheckEndTime()
+	lastCheck, err := scope.LastCheck()
 	if err != nil {
 		if releaseErr := lock.Release(); releaseErr != nil {
 			logger.Error("failed-to-release-lock", releaseErr)
 		}
-
-		return nil, false, fmt.Errorf("get last check end time: %w", err)
+		return nil, false, err
 	}
 
-	runAt := end.Add(interval)
-
 	shouldRun := false
-	if d.build.IsManuallyTriggered() {
-		// ignore interval for manually triggered builds
-		shouldRun = true
-	} else if !d.clock.Now().Before(runAt) {
-		// run if we're past the last check end time
-		shouldRun = true
+	if !d.plan.IsPeriodic() {
+		if !lastCheck.Succeeded || lastCheck.EndTime.Before(d.build.StartTime()) {
+			shouldRun = true
+		}
```

Comment:
Hey @evanchaoli, would you mind explaining what this is for? I'm reading it as: any non-periodic checks (is this just checks for `task.config.image_resource`?) shouldn't run if the prior check succeeded and this build started after it finished. But I'm not totally understanding what this logic is for. Is this to prevent multiple checks for the same `image_resource` from running in the same build?

Reply:
@aoldershaw  Yes, your understanding is correct. When a job has multiple tasks that use same `image_resource`, before this change, it runs a `check` for each task, which is unnecessary. This logic prevents duplicate `check`s for same `image_resource` for the same build. Do you see any problem with this logic?

Reply:
Nope no problem - just wanted to make sure my understanding was correct! Thanks!